src/main/java/hudson/plugins/audit_trail/LogFileAuditLogger.java

```diff
@@ -30,33 +48,143 @@ public class LogFileAuditLogger extends AuditLogger {
+                }
+            }
+        }
+        // Initialize initInstant to the current instant
+        if (initInstant == null) {
+            initInstant = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+        }
         configure();
     }
 
+    private boolean shouldRotate() {
+        return ZonedDateTime.now().isAfter(initInstant.plus(Duration.ofDays(1)));
+    }
+
+    /**
+     * Rotates the daily rotation logger
+     */
+    private void rotate() {
+        if (handler != null) {
+            handler.close();
+        }
+        initInstant = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+        configure();
+        // After rotation we remove old files if needed
+        removeOldFiles();
+    }
+
+    protected String computePattern() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd").withZone(ZoneId.systemDefault());
+        String formattedInstant = formatter.format(initInstant);
+        String computedFileName = String.format("%s-%s", FilenameUtils.getName(basePattern.toString()) , formattedInstant);
+        Path parentFolder = basePattern.getParent();
+        if (parentFolder != null) {
+            return parentFolder.resolve(computedFileName).toString();
+        }
+        return computedFileName;
+    }
+
+    private void removeOldFiles() {
+        Path directoryPath = basePattern.getParent();
+        if (directoryPath != null) {
+            Collection<File> files = FileUtils.listFiles(directoryPath.toFile(), new RegexFileFilter(".*" + FilenameUtils.getName(basePattern.toString()) + DAILY_ROTATING_FILE_REGEX_PATTERN), DirectoryFileFilter.DIRECTORY);
+            if (files.size() > count) {
+                List<File> orderedList = files.stream().sorted(LASTMODIFIED_REVERSE).collect(Collectors.toList());
+                List<File> toDelete = orderedList.subList(count, orderedList.size());
+                for (File file : toDelete) {
+                    if (!file.delete()) {
+                        LOGGER.log(Level.SEVERE, "File {0} could not be removed on rotate overation", file.getName());
+                    }
+                }
+            }
+        }
+    }
+
     @SuppressFBWarnings(
           value="RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE",
           justification = "value can be null if no config file exists")
     private Object readResolve() {
         if(logSeparator == null) {
             logSeparator = DEFAULT_LOG_SEPARATOR;
         }
-        configure();
+        if (rotateDaily) {
+            this.basePattern = Paths.get(log);
+            initializeDailyRotation();
+        } else {
+            configure();
+        }
         return this;
     }
 
     @Override
     public void log(String event) {
+        // to avoid synchronizing the whole method
+        if (rotateDaily && shouldRotate()) {
+            synchronized (monitor) {
```

Comment:
Not sure why you synchronize on `monitor` instead of synchronizing on `this`?

Also I'm under the impression that there is an issue with the logic. Line 183 you are checking if you shouldRotate by reading the content of the `initInstant` field. This field can be updated by another thread in the `rotate` method, therefore you could rotate twice. You probably want to check a second time if you should rotate after acquiring the lock.